### PR TITLE
Support for absolute paths in classpath

### PR DESCRIPTION
**Pull request by hayssams**
Do not prefix with $lib_dir when an absolute path is added to the
classpath

### CODE REVIEW

**Comment by typesafehub-validator** [comment]
Hi @hayssams,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by muuki88** [comment]
Thanks for your pull requests. Can you explain the reasons and use cases? 

In order to accept the pull request we would also need a few things
- Documentation on how to use it
- Tests

**Comment by hayssams** [comment]
When building a package you sometimes need to add in the classpath, an absolute path in your package like a docker volume path  or any external path. 
In my case, I needed to use hive-site.xml present on the target platform and located in an absolute path mounted as a docker volume.

You are right documentation and tests are missing and I have to add them asap.

**Comment by muuki88** [comment]
Still need to get used to the new github review tools ;)

If you need any help on writing tests and documentation you can read the [Developer Guide](https://github.com/sbt/sbt-native-packager/wiki/Developer-Guide) or ask here.

**Comment by typesafehub-validator** [comment]
Hi @hayssams,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by typesafehub-validator** [comment]
Hi @hayssams,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by hayssams** [comment]
@typesafehub-validator CLA already signed

**Comment by hayssams** [comment]
Hi @muuki88 
Just added the test case but can't figure out where the best place is to put the following doc :

To add an absolute path to the app classpath just set it in the scriptClasspath Key as follows :
For Linux: scriptClasspath += "/absolute/path"
For Windows : scriptClasspath += "c:\\absolute\\path"
 Relative path will be prefixed by the app lib directory path

File: src/main/scala/com/typesafe/sbt/packager/archetypes/JavaAppBatScript.scala [hunk 1]
```diff
@@ -11,9 +11,10 @@ object JavaAppBatScript {
 
   def makeWindowsRelativeClasspathDefine(cp: Seq[String]): String = {
     def cleanPath(path: String): String = path.replaceAll("/", "\\\\")
+    def isAbsolute(path: String): Boolean = path.length > "c:\\".length && Character.isLetter(path(0)) && path(1) == ':'
```
Comment (muuki88) [inline src/main/scala/com/typesafe/sbt/packager/archetypes/JavaAppBatScript.scala]: I'm a bit confused about the `"c:\\".length`definition. It looks like putting documentation into the source code.
If you have to check for a path longer than 4 characters do so explicitly and add a small comment for what reason. At first I thought you were checking if the path starts with `c:\\`` ;)

File: src/sbt-test/universal/absolute-path/build.sbt [hunk 2]
```diff
@@ -0,0 +1,21 @@
+enablePlugins(JavaAppPackaging)
```
Comment (muuki88) [inline src/sbt-test/universal/absolute-path/build.sbt]: Test looks good. However can you split this into two tests?

The bash one goes into `sbt-test/bash` and ( I know this is confusing :( ) into `sbt-test/windows`

File: src/main/scala/com/typesafe/sbt/packager/archetypes/JavaAppBatScript.scala (hunk 1 above)
Comment (hayssams) [inline src/main/scala/com/typesafe/sbt/packager/archetypes/JavaAppBatScript.scala]: ok will do it in a few minutes

File: src/sbt-test/universal/absolute-path/build.sbt (hunk 2 above)
Comment (hayssams) [inline src/sbt-test/universal/absolute-path/build.sbt]: ok

Comment (hayssams) [inline src/sbt-test/universal/absolute-path/build.sbt]: done

File: src/main/scala/com/typesafe/sbt/packager/archetypes/JavaAppBatScript.scala (hunk 1 above)
Comment (hayssams) [inline src/main/scala/com/typesafe/sbt/packager/archetypes/JavaAppBatScript.scala]: done

**Comment by muuki88** [comment]
Awesome. Thanks :)

**Pull request merged**